internal/access/credential.go

```diff
@@ -225,3 +235,59 @@ func checkPasswordRequirements(db data.ReadTxn, password string) error {
 
 	return nil
 }
+
+var badPasswordsFile string = path.Join(os.TempDir(), "bad-passwords")
+
+// checkBadPasswords checks if the password is a known bad password, i.e. a widely reused password.
+// It checks the password against a file on disk. If the file is empty, it will download and populate
+// the file from Have I Been Pwned's top 100,000 most common passwords.
+func checkBadPasswords(password string) error {
+	file, err := os.OpenFile(badPasswordsFile, os.O_RDWR|os.O_CREATE, 0o600)
+	if err != nil {
+		return err
+	}
+
+	stat, err := file.Stat()
+	if err != nil {
+		return err
+	}
+
+	if stat.Size() == 0 {
+		request, err := http.NewRequestWithContext(context.Background(), http.MethodGet, "https://www.ncsc.gov.uk/static-assets/documents/PwnedPasswordsTop100k.txt", nil)
```

Comment:
This downloads the bad password list and stores it in the filesystem if it doesn't already exist. Alternatively, it can be stored in Redis which would make retrieval trivial but adds another dependency on Redis

Reply:
Any reason we can't embed this file rather than downloading it?